sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,1030 @@
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::GetProperties
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/getproperties
+   */
+  struct PathGetPropertiesOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief If the value is PathGetPropertiesAction::GetStatus only the system defined properties
+     *        for the path are returned. If the value is PathGetPropertiesAction::GetAccessControl
+     *        the access control list is returned in the response headers (Hierarchical Namespace
+     *        must be enabled for the account), otherwise the properties are returned.
+     */
+    PathGetPropertiesAction Action = PathGetPropertiesAction::Unknown;
+
+    /**
+     * @brief Valid only when Hierarchical Namespace is enabled for the account. If "true",
+     *        the user identity values returned in the x-ms-owner, x-ms-group, and x-ms-acl
+     *        response headers will be transformed from Azure Active Directory Object IDs to
+     *        User Principal Names. If "false", the values will be returned as Azure Active
+     *        Directory Object IDs. The default value is false. Note that group and application
+     *        Object IDs are not translated because they do not have unique friendly names.
+     */
+    bool UserPrincipalName = bool();
+
+    /**
+     * @brief If specified, the operation only succeeds if the resource's lease is active and
+     *        matches this ID.
+     */
+    std::string LeaseId;
+
+    /**
+     * @brief Specify an ETag value to operate only on path with a matching value.
+     */
+    std::string IfMatch;
+
+    /**
+     * @brief Specify an ETag value to operate only on path without a matching value.
+     */
+    std::string IfNoneMatch;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has been modified since the
+     (        specified date and time.
+    */
+    std::string IfModifiedSince;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::Lease
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/lease
+   */
+  struct PathLeaseOptions : public SharedRequestOptions
```

Comment:
There will be a `LeaseClient` and several functions for different lease operations. So i guess there will also be different lease options.

Reply:
We can discuss on this internally first. I personally don't like the way lease client works.

Reply:
Added a design topic to discuss.